### PR TITLE
chore(cluster_agent): added logs for get nvme controller

### PR DESCRIPTION
**Pull request by abhilashshetty04**
Prinitng err returned by get_nvme_controller call

### CODE REVIEW

File: control-plane/agents/src/bin/ha/cluster/switchover.rs [hunk 1]
```diff
@@ -237,7 +237,8 @@ impl SwitchOverRequest {
                                 info!("Retrying Republish after cleaning up targets");
                                 return Ok(());
                             }
-                            _ => {
+                            Err(e) => {
+                                error!("error while listing nvme controller {}", e.to_string());
```
Comment (tiagolobocastro) [inline control-plane/agents/src/bin/ha/cluster/switchover.rs]: instead, please add instrument at the top of the function:
#[tracing::instrument(level = "trace", skip(self), fields(volume.uuid = %self.volume_uuid), error]

File: control-plane/agents/src/bin/ha/cluster/switchover.rs [hunk 2]
```diff
@@ -237,8 +238,7 @@ impl SwitchOverRequest {
                                 info!("Retrying Republish after cleaning up targets");
                                 return Ok(());
                             }
-                            Err(e) => {
-                                error!("error while listing nvme controller {}", e.to_string());
+                            _ => {
```
Comment (tiagolobocastro) [inline control-plane/agents/src/bin/ha/cluster/switchover.rs]: please also include error.to_string() or at least err.kind in the error we return below

File: control-plane/agents/src/bin/ha/cluster/switchover.rs (hunk 1 above)
Comment (abhilashshetty04) [reply control-plane/agents/src/bin/ha/cluster/switchover.rs]: Done

File: control-plane/agents/src/bin/ha/cluster/switchover.rs (hunk 2 above)
Comment (abhilashshetty04) [reply control-plane/agents/src/bin/ha/cluster/switchover.rs]: Done

File: control-plane/agents/src/bin/ha/cluster/switchover.rs [hunk 3]
```diff
@@ -237,9 +238,11 @@ impl SwitchOverRequest {
                                 info!("Retrying Republish after cleaning up targets");
                                 return Ok(());
                             }
-                            _ => {
-                                let err =
-                                    "Failed to list Nvme subsystems for the volume".to_string();
+                            Err(e) => {
+                                let err = format!(
+                                    "Failed to list Nvme subsystems for the volume.err kind: {:?}",
```
Comment (tiagolobocastro) [inline control-plane/agents/src/bin/ha/cluster/switchover.rs]:
```suggestion
                                    "Failed to list Nvme subsystems for the volume, error kind: {:?}",
```

File: control-plane/agents/src/bin/ha/cluster/switchover.rs (hunk 1 above)
Comment (tiagolobocastro) [reply control-plane/agents/src/bin/ha/cluster/switchover.rs]: Actually use level = "info" as it'll be useful to see the spans on jaeger

**Comment by abhilashshetty04** [comment]
bors try

File: control-plane/agents/src/bin/ha/cluster/switchover.rs [hunk 4]
```diff
@@ -237,9 +238,10 @@ impl SwitchOverRequest {
                                 info!("Retrying Republish after cleaning up targets");
                                 return Ok(());
                             }
-                            _ => {
-                                let err =
-                                    "Failed to list Nvme subsystems for the volume".to_string();
+                            Err(error) => {
+                                let err = format!(
+                                    "Failed to list Nvme subsystems for the volume, error : {error}"
```
Comment (tiagolobocastro) [inline control-plane/agents/src/bin/ha/cluster/switchover.rs]: Spurious space:

```suggestion
                                    "Failed to list Nvme subsystems for the volume, error: {error}"
```

**Review by tiagolobocastro — APPROVED** [review]

**Comment by abhilashshetty04** [comment]
bors merge

**Pull request merged**